python/pyarrow/array.pxi

```diff
@@ -986,14 +986,49 @@ cdef class Array(_PandasConvertible):
     @property
     def nbytes(self):
         """
-        Total number of bytes consumed by the elements of the array.
+        Returns the sum of bytes from all buffer ranges referenced
```

Comment:
I personally find the original text clearer. We can maybe leave it as is, and add the "buffer ranges referenced" in the next paragraph below?